### PR TITLE
Added rule and tidied up a bit

### LINKED ISSUE #3811: Update style guide: new rules for using code formatting
We have decided on new style rules. Add them to the style guide.

- [x] **Add rule: "Do not use code in headings"**

     Suggested location for rule: https://github.com/knative/docs/blob/mkdocs/docs/help/contributor/style-guide/style-and-formatting.md. 
    Decided in this [Slack thread](https://knative.slack.com/archives/C9CV04DNJ/p1623754932034700)

- [ ] **Add to list of things that should be in code**

    Add the following items:
    - CLI names eg `kn` and `kubectl`
    - Fields or properties from a YAML file

    Location for rule: https://github.com/knative/docs/blob/mkdocs/docs/help/contributor/style-guide/documenting-code.md
    Decided in this [Slack thread](https://knative.slack.com/archives/C9CV04DNJ/p1623678061012800?thread_ts=1623674397.007600&cid=C9CV04DNJ)

### PR DESCRIPTION
**Pull request by RichardJJG**
Fixes #3811

### CODE REVIEW

**Comment by RichardJJG** [comment]
Outside of issue scope:
* I re-organized the contents of the Style and Formatting.
* I added another heading rule that already exists in the Procedure template: use imperatives for procedure headings. Not within this issue’s scope, I know, but it’s a small uncontroversial change I just wanted to squeeze in.

File: docs/help/contributor/style-guide/style-and-formatting.md [hunk 1]
```diff
@@ -51,14 +53,36 @@ between the number and the unit.
...
+
+>Put an acronym inside parentheses after its explanation. Don’t use parentheses for anything else.
+
+>Parenthetical statements especially should be avoided because readers skip them.
+If something is important enough to be in the sentence, it should be fully part of that sentence.
+
+|:white_check_mark: Correct                  |:no_entry: Incorrect
+|--------------------|-----
+|Custom Resource Definition (CRD) |Check your CLI (you should see it there)
+|Knative Serving will create a Revision |Knative will create a Revision (a stateless, snapshot in time of your code and configuration)|
```
Comment (abrennan89) [inline docs/help/contributor/style-guide/style-and-formatting.md:83]:
```suggestion
|Knative Serving creates a Revision |Knative creates a Revision (a stateless, snapshot in time of your code and configuration)|
```
Nitpicking, but I'd prefer present tense:

https://kubernetes.io/docs/contribute/style/_print/#use-present-tense)
https://developers.google.com/style/tense?hl=en

Maybe we should add something about tenses if it's not in the guide already?

Comment (abrennan89) [reply docs/help/contributor/style-guide/style-and-formatting.md:83]: Oh, nvm it's here 😄 https://github.com/knative/docs/pull/4035/files#diff-0138e4edfa32eb45b8ff7c58d77a845112e9bf6b6694af396b4cf53b44fc8bc4L3

**Comment by RichardJJG** [comment]
Thanks for catching the stray future tense, @abrennan89 -- you can lgtm now

**Review by abrennan89 — APPROVED** [review]
/lgtm

**Pull request merged**